zone/client_packet.cpp

```diff
@@ -16183,3 +16343,45 @@ void Client::RecordKilledNPCEvent(NPC *n)
 		}
 	}
 }
+
+void Client::Handle_OP_RaidDelegateAbility(const EQApplicationPacket* app) 
+{
+	if (app->size != sizeof(DelegateAbility_Struct)) {
+		LogDebug("Size mismatch in OP_RaidDelegateAbility expected [{}] got [{}]", sizeof(DelegateAbility_Struct), app->size);
+		DumpPacket(app);
+		return;
+	}
+
+	DelegateAbility_Struct* das = (DelegateAbility_Struct*)app->pBuffer;
+
+	switch (das->DelegateAbility)
+	{
+	case RaidDelegateMainAssist: 
+	{
+		auto r = entity_list.GetRaidByName(this->GetName());
+		r->DelegateAbilityAssist(this, das->Name);
+		break;
+	}
+	case RaidDelegateMainMarker: 
+	{
+		auto r = entity_list.GetRaidByName(this->GetName());
+		r->DelegateAbilityMark(this, das->Name);
+		break;
+	}
+	default:
+		LogDebug("RaidDelegateAbility default case");
+		break;
+	}
+}
+
+void Client::Handle_OP_RaidClearNPCMarks(const EQApplicationPacket* app)
+{
+	if (app->size != 0)	{
+		LogDebug("Size mismatch in OP_RaidClearNPCMark expected [{}] got [{}]", 0, app->size);
+		DumpPacket(app);
+		return;
+	}
+
+	auto r = entity_list.GetRaidByName(GetName());
+	r->RaidClearNPCMarks(GetName());
```

Comment:
You should pass `Client *c` here and use it as such inside the function

Example

```cpp
r->RaidClearNPCMarks(this);
```

```cpp
void Raid::RaidClearNPCMarks(Client *c) 
```

```cpp
void Raid::RaidClearNPCMarks(Client *c) 
{
	// remove this since you already have the client
	// auto mob_id = entity_list.GetMob(client_name)->GetID();
	// ...truncated
        entity_list.GetClientByID(mob_id)->MessageString(Chat::Cyan, NOT_DELEGATED_MARKER);
        // to
        c->MessageString(Chat::Cyan, NOT_DELEGATED_MARKER);
```

Code like this in the function

```cpp
strcasecmp(main_marker_pcs[MAIN_MARKER_1_SLOT], client_name)
```

Can become

```cpp
strcasecmp(main_marker_pcs[MAIN_MARKER_1_SLOT], c->GetCleanName())
```

Reply:
Updated